### PR TITLE
added confirmation before deleting submissions, users and problems

### LINKED ISSUE #72: Add confirmation before deleting submissions, users, and problems
## Summary

For admins when selecting problems, submissions, and users to delete prompt them "Are you sure?" including a disclaimer about what else will be deleted (e.j with problems submissions will be deleted, and with users submissions will be deleted)

### Basic example

Sweet alert could be used

### PR DESCRIPTION
**Pull request by MrKevinOConnell**
# Description

- confirmation window shows before users, submissions and problems are deleted
- upon the deletion occurring successfully a toast container shows
- I want to add a follow up issue for someone else to do for displaying a error message

Fixes #72 

## Type of change

- [x] 💡 New feature

# How Has This Been Tested?

- select a user on the table, click delete selected

Screen Shot 2021-10-08 at 1 36 02 PM

- select ok, the user should be deleted, and a message should be displayed confirming the deletion occurred

Screen Shot 2021-10-08 at 1 39 15 PM

- [x] Tested individually.

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] My changes do not break any features.

### CODE REVIEW

**Comment by MrKevinOConnell** [comment]
@Jl5her whenever the server returns a call we prompt toast by sending a toast.success or toast.error depending on the result (which gets sent to the container, makes it a lot easier to manage notifications

File: frontend/src/pages/admin/submissions/Submissions.tsx [hunk 1]
```diff
@@ -82,22 +82,33 @@ const Submissions = (): JSX.Element => {
...
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${localStorage.accessToken}`
+        },
+        body: JSON.stringify({ sid: submissionsToDelete })
+      })
+      if (response.ok) {
+        loadSubmissions()
+        //tells the toast container below to display a message saying 'Deleted selected submissions'
+        const id = submissionsToDelete.join()
```
Comment (a1g3) [inline frontend/src/pages/admin/submissions/Submissions.tsx:102]: Why do the submissions have to be joined to form the `id`?

File: frontend/src/pages/admin/users/Users.tsx [hunk 2]
```diff
@@ -144,35 +144,45 @@ const Users = (): JSX.Element => {
...
-      method: 'DELETE',
-      headers: {
-        'Content-Type': 'application/json',
-        Authorization: `Bearer ${localStorage.accessToken}`
-      },
-      body: JSON.stringify({ uid: usersToDelete })
-    })
+      if (response.ok) {
+        setUsers(users.filter((user) => !usersToDelete.includes(user.uid)))
+        const id = usersToDelete.join()
```
Comment (a1g3) [inline frontend/src/pages/admin/users/Users.tsx:168]: See my previous comment.

**Review by a1g3 — CHANGES_REQUESTED** [review]
Make sure you've also implemented the changes Jack P. requested.

**Review by JForden — DISMISSED** [review]

File: frontend/src/pages/admin/submissions/Submissions.tsx (hunk 1 above)
Comment (MrKevinOConnell) [reply frontend/src/pages/admin/submissions/Submissions.tsx:102]: not sure, that was something that jack made

**Pull request merged**